src/Uplink.php

```diff
@@ -35,10 +35,39 @@ public static function create(): self
         $arch = php_uname('m');
         $os = strtolower(php_uname('s'));
 
-        $ffi = FFI::cdef(
-            file_get_contents($root . '/build/uplink-php.h'),
-            "{$root}/build/libuplink-{$arch}-{$os}.so"
-        );
+        $libuplinkSo = "{$root}/build/libuplink-{$arch}-{$os}.so";
+        try {
+            $ffi = FFI::cdef(
+                file_get_contents($root . '/build/uplink-php.h'),
+                $libuplinkSo
+            );
+        } catch (\FFI\Exception $ffiException) {
+            // dlerror() may have a more detailed error.
+            // Example: no permission, wrong glibc version, unknown binary format etc.
+            // This check should really be in PHP core.
+            try {
+                $ffi = FFI::cdef("
+                    void *dlopen(const char *filename, int flags);
+                    char *dlerror();
+                ");
+                $ffi->dlopen($libuplinkSo, 1);
```

Comment:
do we need to `dlclose` here as well?

Reply:
Well I expect the call to fail. But I've added it for completeness.